### ISSUE TITLE
JAVADOC_AUTOBRIEF and @File regression

### ISSUE THREAD

**Issue by rodanrider** [doxygen/doxygen#7867]
**Describe the bug**
JAVADOC_AUTOBRIEF does not work as expected for @file documentation since v1.8.16.  Earlier releases work fine.

**Expected behavior**
I expect that the first line of text after a @file tag would be treated as the @brief description in a @file block.  Something changed in v1.8.16 and now even with the JAVADOC_AUTOBRIEF = YES, an explicit @brief tag is required.

**Screenshots**
v1.8.15 - Expected behaviour:
![image](https://user-images.githubusercontent.com/57236581/85332001-fad5f080-b48b-11ea-89dd-d0681a37d25c.png)

v1.8.16 - Unexpected behaviour:
![image](https://user-images.githubusercontent.com/57236581/85332045-104b1a80-b48c-11ea-9b14-624c96afd266.png)

**To Reproduce**
This is a default configuration file with the following 2 modifications:
JAVADOC_AUTOBRIEF = YES
OPTIMIZE_OUTPUT_FOR_C = YES

This is a pretty simple test case:
```
/**
 * @file test.h
 * This is a brief description.
 */
```
[test_case.zip](https://github.com/doxygen/doxygen/files/4815444/test_case.zip)

**Version**
I am using release versions of v1.8.15 - Expected behaviour, and v1.8.16 - Unexpected behaviour.
I have also confirmed that the released v1.8.18 exhibits the same buggy behaviour.

**Stack trace**
N/A

**Additional context**
N/A

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7867

**Comment by rodanrider** [comment]
Thanks for the quick fix.  It fixes my reduced test case.  However, it causes other problems, such as if another tag is present after the file tag.  So I would suggest cancelling your pull request until those are also resolved.

I built it with the latest commit d5916c06fba9667911a19cf658e7c5ab54daf72c + your pull request.

```
/**
 * @file test_file_author_brief.h
 * @author John Doe
 *
 * This is a brief description.
 */
```
[v1_8_19_fix_bad.zip](https://github.com/doxygen/doxygen/files/4839388/v1_8_19_fix_bad.zip)

The author tag is used as the brief description!

![image](https://user-images.githubusercontent.com/57236581/85904351-591d0f00-b7bd-11ea-998e-d6012ee5019d.png)

**Comment by albert-github** [comment]
Thanks for the second test case, this looks worse than the original problem, I'll have to dive into it.

**Comment by albert-github** [comment]
I have withdrawn the proposed pull request #7868 as most likely another strategy has to be chosen.

Probably something along the lines:
As long as no plain text is given and the given commands are "brief documentation terminating" these commands land in their normal places and after each "end" it is checked whether or not the the next paragraph can be a brief description or not. The reset should probably not take place in case of commands like `\brief`, `\detaild`, `\copydoc`, `\copybrief`, `\copydetails` and ...

**Comment by doxygen** [comment]
Another idea would be to add a column to docCmdMap with an extra attribute telling if it is a "structure indicating command" (like \class, \struct, \var, \def, \fn, etc) or some other command. In case of a structure command it should do this "special" conditional evaluation.

I think we should also consider at this case

```{c}
/**
 * @file file1.cpp
 * Brief description
 *
 * Details
 *
 * @file file2.cpp
 * Brief or defailed description?
 *
 * Details
 *
 * @fn foo()
 * Brief or detailed for foo?
 *
 * Details
 *
 * @fn bar()
 * Brief or detailed for bar?
 *
 * Details
 */

```

would make sense to switch back to brief description for the second `@file` and for each of the `@fn`s. This was the behavior prior to 1.8.16. Maybe also good to look what commit caused the change in behavior.

**Comment by albert-github** [comment]
I definitely think it would be a good idea to have an extra column with the docCmdMap as this would make it easier to control.
Besides already mentioned command I'm also thinking about commands like `\mainpage` / `\page`/ `\section` ...; `\par`.

The change in behavior most lkely came from the fact that a number of commands have been set to "brief terminating" as they were not meant to be used in a "brief description".

**Comment by doxygen** [comment]
> I definitely think it would be a good idea to have an extra column with the docCmdMap as this would make it easier to control.
> Besides already mentioned command I'm also thinking about commands like `\mainpage` / `\page`/ `\section` ...; `\par`.
> 

`\section` and `\par` probably not as these actual add content to the comment block; in general all commands for which `makeStructuralIndicator()` is called, which indeed includes `\page` and `\mainpage`.

> The change in behavior most lkely came from the fact that a number of commands have been set to "brief terminating" as they were not meant to be used in a "brief description".

Indeed, but without distinguishing between commands that do not add content to a command block but only indicate where a comment block should end up and commands that do add content that does not fit into a brief description (like `\author`)

**Comment by doxygen** [comment]
Note that this issue is related to / a duplicate of #7852

**Comment by albert-github** [comment]
The place for setting the switch to "non brief documentation" is clear and could be similar as proposed in the pull request #7868 with:
- should also be valid for commands without a handler function
- the fact that a switch has been made should probably be recorded so it can be reset later

The question for me is where has the reset to take place.

**Comment by albert-github** [comment]
A patch (commit: 327423e217337bf4e64515aba6cf78b9877bc165) has been integrated into the master version.

I still see that the "bad fix example" (see comment https://github.com/doxygen/doxygen/issues/7867#issuecomment-650422682), does not give a brief description but leaves it empty.
I don't know whether this is a special case or not / the exact rationale for this.
I tried:
- removing empty line: no improvement
- changing author command internally from `Block` to `Invisible` or `Inline`: bad idea as now the problem of the previous fix reappeared
- adding `\brief`: did work.

**Comment by albert-github** [comment]
Looks like the problem as indicated in #6945 is back again.
```
/*! \brief Spawns two mettaurs on the field.
 * \class TwoMettaurMob
 *
 * May optionally spawn an empty tile for challenge
 */

#pragma once
#include "bnMobFactory.h"
#include "bnMettaur.h"
#include "bnMettaurIdleState.h"

class TwoMettaurMob :
  public MobFactory
{
public:
  /** constructor */
  TwoMettaurMob(Field* field);
  /** destructor */
  ~TwoMettaurMob();

  /**
   * @brief Builds and returns the mob
   * @return Mob pointer. must be deleted manually.
   */
  Mob* Build();

  /**
   * @brief Builds and returns the mob
   */
  Mob* Build(int m);
};
```

Results in a brief file description of: 
> Spawns two mettaurs on the field.May optionally spawn an empty tile for challenge 

Looking at the code in `commentscan.l` I don't see a distinction between the internal states `Inline` and `Invisible`.

**Comment by doxygen** [comment]
@albert-github thanks for the report. I already feared my fix was too simple ☹️. In your example the problem is that the newline after `\class` is consumed by the command; and then the blank line after it is no longer seen as a paragraph separator. I'll unput the newline hoping this doesn't have any negative side-effects.

For now there is indeed no difference between `Inline` and `Invisible`, but maybe this could still be useful later.

**Comment by albert-github** [comment]
When having the following example (with default Doxyfile):
```
/*! \file
 * \brief Spawns third mettaurs on the field \defgroup Group1 Group 1 desc
 * The brief of the group 1
 */
```

we get in case if 1.8.18:
![res_1_8_18](https://user-images.githubusercontent.com/5223533/87858125-e3274600-c92b-11ea-8f13-a47783d083d2.jpg)

but with the new version (1fb2408b which incorporates the actual change 7fa193c):
![res_1fb2408b](https://user-images.githubusercontent.com/5223533/87858121-df93bf00-c92b-11ea-9037-be7750b25005.jpg)

So looks like we lost the brief description for the group.

**Comment by doxygen** [comment]
> When having the following example (with default Doxyfile):
> 
> ```
> /*! \file
>  * \brief Spawns third mettaurs on the field \defgroup Group1 Group 1 desc
>  * The brief of the group 1
>  */
> ```
> 
> 
> So looks like we lost the brief description for the group.

I also don't see a brief description for the group in the example.
It can be activated by enabling `QT_AUTOBRIEF`.
So I think the behavior introduced in version 1.8.16 was actually wrong.

**Comment by albert-github** [comment]
I think that makes sense.

A bit strange though, for me, is why there is no brief description with the class, constructor, destructor and second Build(int) function in:
```/*! \file
 * \brief Spawns third mettaurs on the field \defgroup Group Group desc
 * The brief of the group
 */

  /**
   * \fn ThirdMettaurMob::Build(int)
   * Builds and returns the mob, initialized
   */

#pragma once
#include "bnMobFactory.h"
#include "bnMettaur.h"
#include "bnMettaurIdleState.h"

/** the third class */
class ThirdMettaurMob :
  public MobFactory
{
public:
  /** constructor */
  ThirdMettaurMob(Field* field);
  /** destructor */
  ~ThirdMettaurMob();

  /**
   * @brief Builds and returns the mob
   * @return Mob pointer. must be deleted manually.
   */
  Mob* Build();

  Mob* Build(int m);
};
```
and settings:
```
QUIET=YES
QT_AUTOBRIEF=YES
```

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
